### PR TITLE
Initial draft of clock support

### PR DESCRIPTION
**Pull request by pmai**

### CODE REVIEW

**Comment by JochenKoehler** [comment]
Web meeting 2023-11-24
Ulrich, Klaus will review this pull request

**Comment by JochenKoehler** [comment]
Web meeting 2024-03-15:
Some documentation should be added --> Pierre will prepare a draft

File: docs/5___ssd.adoc [hunk 1]
```diff
@@ -137,15 +137,18 @@ The following XML child elements are specified for the Connector element:
...
+|Clock |One or more of these optional elements associate the connector to a clock of the given name, which must be defined on the element that contains this connector.
 |ConnectorGeometry |This optional element defines the geometry information of the connector. See below for details.
 |===
 
-The type of the Connector is identified by the presence of one of the XML child elements Real, Float64, Float32, Integer, Int8, UInt8, Int16, UInt16, Int32, UInt32, Int64, UInt64, Boolean, String, Enumeration, or Binary.
+The type of the Connector is identified by the presence of one of the XML child elements Real, Float64, Float32, Integer, Int8, UInt8, Int16, UInt16, Int32, UInt32, Int64, UInt64, Boolean, String, Enumeration, Binary, or Clock.
 
 The dimensionality of the Connector is given by the presence of one or more Dimension elements.
 
+The association of a conenctor to a clock is given by the presence of one or more Clock elements.
```
Comment (klausschuch) [inline docs/5___ssd.adoc:150]:
```suggestion
The association of a connector to a clock is given by the presence of one or more Clock elements.
```

**Review by masoud-najafi — APPROVED** [review]
Just a short comment about this phrase: "It is implementation-defined whether and which connections between two connectors of kind `input` with `intervalVariability` that is not `triggered` are allowed." This leaves the door open for different interpretations about connecting these "time-based" clocks. For example as you indicated for establishing a constraint equality or have a union of several clock ticks. This would make an SSP model with such connections completely tool dependent.

**Comment by pmai** [comment]
> Just a short comment about this phrase: "It is implementation-defined whether and which connections between two connectors of kind `input` with `intervalVariability` that is not `triggered` are allowed." This leaves the door open for different interpretations about connecting these "time-based" clocks. For example as you indicated for establishing a constraint equality or have a union of several clock ticks. This would make an SSP model with such connections completely tool dependent.

Indeed it does, however as there is currently no established practice on how to handle this across tools, I think this is currently the only thing we can do. If common or multiple common practice is established, we can either tighten the requirements in future releases, and/or add additional attributes to select between multiple ways.

**Comment by JochenKoehler** [comment]
Web meeting 2024-05-08
Pierre will do the merge

**Pull request merged**